lib/Change.lua

```diff
@@ -0,0 +1,23 @@
+local Change = {}
+
+local changeMetatable = {
+	__tostring = function(self)
+		return ("ChangeListener(%s)"):format(self.name)
+	end
+}
+
+setmetatable(Change, {
+	__index = function(self, propertyName)
+		local changeListener = {
+			type = "propertyChange",
```

Comment:
We should set the type to `Change` to line up with how `Event` works.

This also lines up really nicely with the _other_ changes I want to make to typecheck arguments using the `type` value!

Reply:
Oops, I missed that part of the refactoring! Fixed now.